go/storage/client/client.go

```diff
@@ -304,10 +304,26 @@ func (b *storageClientBackend) readWithClient(
 		// from the connected nodes:
 		// https://github.com/oasisprotocol/oasis-core/issues/1815.
 		rng := rand.New(mathrand.New(cryptorand.Reader))
+		nodeOrder := rng.Perm(n)
+
+		// If a storage node priority hint is set, prioritize overlapping nodes.
+		if nodes := api.NodePriorityHintFromContext(ctx); len(nodes) > 0 {
+			var indices []int
+			for _, nodeID := range nodes {
+				for i, c := range conns {
+					if c.Node.ID.Equal(nodeID) {
+						indices = append(indices, i)
+						break
+					}
+				}
+			}
+
+			nodeOrder = append(indices, nodeOrder...)
```

Comment:
This will result in duplicates, so if a connection to a "priority" node is not working (or the request errors for whatever reason), it can get queried twice. Which is technically ok, but might be nicer cleaning this of duplicates or at least adding a note about it.

Reply:
Yeah I thought about it, but it didn't seem to be worth the added complexity. I'll think about it some more :)